### ISSUE TITLE
pls publish a new release since v10.24.0 has too many CVEs

### ISSUE THREAD

**Issue by andyzhangx** [Azure/azure-storage-azcopy#2675]
##### Note: The version is visible when running AzCopy without any argument
v10.24.0

### Which platform are you using? (ex: Windows, Mac, Linux)
Linux

### What problem was encountered?

```
usr/local/bin/azcopy (gobinary)
===============================
Total: 13 (UNKNOWN: 1, MEDIUM: 9, HIGH: 3, CRITICAL: 0)

┌─────────┬────────────────┬──────────┬────────┬───────────────────┬──────────────────────────────────┬──────────────────────────────────────────────────────────────┐
│ Library │ Vulnerability  │ Severity │ Status │ Installed Version │          Fixed Version           │                            Title                             │
├─────────┼────────────────┼──────────┼────────┼───────────────────┼──────────────────────────────────┼──────────────────────────────────────────────────────────────┤
│ stdlib  │ CVE-2023-39325 │ HIGH     │ fixed  │ 1.19.12           │ 1.20.10, 1.21.3                  │ golang: net/http, x/net/http2: rapid stream resets can cause │
│         │                │          │        │                   │                                  │ excessive work (CVE-2023-44487)                              │
│         │                │          │        │                   │                                  │ https://avd.aquasec.com/nvd/cve-2023-39325                   │
│         ├────────────────┤          │        │                   ├──────────────────────────────────┼──────────────────────────────────────────────────────────────┤
│         │ CVE-2023-45283 │          │        │                   │ 1.20.11, 1.21.4, 1.20.12, 1.21.5 │ The filepath package does not recognize paths with a \??\    │
│         │                │          │        │                   │                                  │ prefix as...                                                 │
│         │                │          │        │                   │                                  │ https://avd.aquasec.com/nvd/cve-2023-45283                   │
│         ├────────────────┤          │        │                   ├──────────────────────────────────┼──────────────────────────────────────────────────────────────┤
│         │ CVE-2023-45287 │          │        │                   │ 1.20.0                           │ golang: crypto/tls: Timing Side Channel attack in RSA based  │
│         │                │          │        │                   │                                  │ TLS key exchanges....                                        │
│         │                │          │        │                   │                                  │ https://avd.aquasec.com/nvd/cve-2023-45287                   │
│         ├────────────────┼──────────┤        │                   ├──────────────────────────────────┼──────────────────────────────────────────────────────────────┤
│         │ CVE-2023-39318 │ MEDIUM   │        │                   │ 1.20.8, 1.21.1                   │ golang: html/template: improper handling of HTML-like        │
│         │                │          │        │                   │                                  │ comments within script contexts                              │
│         │                │          │        │                   │                                  │ https://avd.aquasec.com/nvd/cve-2023-39318                   │
│         ├────────────────┤          │        │                   │                                  ├─────────────────────────────────���────────────────────────────┤
│         │ CVE-2023-39319 │          │        │                   │                                  │ golang: html/template: improper handling of special tags     │
│         │                │          │        │                   │                                  │ within script contexts                                       │
│         │                │          │        │                   │                                  │ https://avd.aquasec.com/nvd/cve-2023-39319                   │
│         ├────────────────┤          │        │                   ├──────────────────────────────────┼──────────────────────────────────────────────────────────────┤
│         │ CVE-2023-39326 │          │        │                   │ 1.20.12, 1.21.5                  │ golang: net/http/internal: Denial of Service (DoS) via       │
│         │                │          │        │                   │                                  │ Resource Consumption via HTTP requests...                    │
│         │                │          │        │                   │                                  │ https://avd.aquasec.com/nvd/cve-2023-39326                   │
│         ├────────────────┤          │        │                   ├──────────────────────────────────┼──────────────────────────────────────────────────────────────┤
│         │ CVE-2023-45284 │          │        │                   │ 1.20.11, 1.21.4                  │ On Windows, The IsLocal function does not correctly detect   │
│         │                │          │        │                   │                                  │ reserved de ......                                           │
│         │                │          │        │                   │                                  │ https://avd.aquasec.com/nvd/cve-2023-45284                   │
│         ├────────────────┤          │        │                   ├──────────────────────────────────┼──────────────────────────────────────────────────────────────┤
│         │ CVE-2023-45289 │          │        │                   │ 1.21.8, 1.22.1                   │ golang: net/http/cookiejar: incorrect forwarding of          │
│         │                │          │        │                   │                                  │ sensitive headers and cookies on HTTP redirect...            │
│         │                │          │        │                   │                                  │ https://avd.aquasec.com/nvd/cve-2023-45289                   │
│         ├────────────────┤          │        │                   │                                  ├──────────────────────────────────────────────────────────────┤
│         │ CVE-2023-45290 │          │        │                   │                                  │ golang: net/http: memory exhaustion in                       │
│         │                │          │        │                   │                                  │ Request.ParseMultipartForm                                   │
│         │                │          │        │                   │                                  │ https://avd.aquasec.com/nvd/cve-2023-45290                   │
│         ├────────────────┤          │        │                   │                                  ├──────────────────────────────────────────────────────────────┤
│         │ CVE-2024-24[78](https://github.com/kubernetes-sigs/blob-csi-driver/actions/runs/8996915504/job/24714144986#step:6:79)3 │          │        │                   │                                  │ golang: crypto/x509: Verify panics on certificates with an   │
│         │                │          │        │                   │                                  │ unknown public key algorithm...                              │
│         │                │          │        │                   │                                  │ https://avd.aquasec.com/nvd/cve-2024-247[83](https://github.com/kubernetes-sigs/blob-csi-driver/actions/runs/8996915504/job/24714144986#step:6:84)                   │
│         ├────────────────┤          │        │                   │                                  ├──────────────────────────────────────────────────────────────┤
│         │ CVE-2024-247[84](https://github.com/kubernetes-sigs/blob-csi-driver/actions/runs/8996915504/job/24714144986#step:6:85) │          │        │                   │                                  │ golang: net/mail: comments in display names are incorrectly  │
│         │                │          │        │                   │                                  │ handled                                                      │
│         │                │          │        │                   │                                  │ https://avd.aquasec.com/nvd/cve-2024-24784                   │
│         ├────────────────┤          │        │                   │                                  ├──────────────────────────────────────────────────────────────┤
│         │ CVE-2024-247[85](https://github.com/kubernetes-sigs/blob-csi-driver/actions/runs/8996915504/job/24714144986#step:6:86) │          │        │                   │                                  │ golang: html/template: errors returned from MarshalJSON      │
│         │                │          │        │                   │                                  │ methods may break template escaping                          │
│         │                │          │        │                   │                                  │ https://avd.aquasec.com/nvd/cve-2024-24785                   │
│         ├────────────────┼──────────┤        │                   ├──────────────────────────────────┼──────────────────────────────────────────────────────────────┤
│         │ CVE-2024-247[88](https://github.com/kubernetes-sigs/blob-csi-driver/actions/runs/8996915504/job/24714144986#step:6:89) │ UNKNOWN  │        │                   │ 1.21.10, 1.22.3                  │ https://avd.aquasec.com/nvd/cve-2024-24788                   │
└─────────┴────────────────┴──────────┴────────┴───────────────────┴──────────────────────────────────┴──────────────────────────────────────────────────────────────┘
```

**Comment by andyzhangx** [comment]
that's the stdlib CVE related to golang version, pls use go v1.22.3 to build azcopy binary, and publish new release, thanks.

**Comment by andyzhangx** [comment]
cc @gapra-msft

**Comment by gapra-msft** [comment]
Hi @andyzhangx we have already released a preview of 10.25.0 a month ago which should resolve this issue. Here is the PR, we are now building with 1.21.8 https://github.com/Azure/azure-storage-azcopy/commit/99ea34db7d7d5362f98cdb1214a9930b51aaa6c2

It looks like we might need to bump up the version to 1.22.1+ so we will take up an item to get that bumped further before the 10.25.0 GA release.

**Comment by andyzhangx** [comment]
@gapra-msft pls bump golang version to v1.22.3, and then publish any new release, thanks.

**Comment by andyzhangx** [comment]
@gapra-msft could you cut a new release building with go v1.22.3  this week?

**Comment by seanmcc-msft** [comment]
@andyzhangx, we our next release will be 1.23, ideally next week.  We don't have the bandwidth for a patch release.

**Comment by andyzhangx** [comment]
@seanmcc-msft thanks for the reply, pls close this issue as long as the new release is published. we aks team are suffered from this azcopy cve, and once you have published a new release, I will cut aks csi drivers asap, thanks.

**Comment by andyzhangx** [comment]
10.25.0 is not published yet

**Comment by adreed-msft** [comment]
10.25 is now released, based on Go 1.22.3
